cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/io/cdap/cdap/datapipeline/StageOperationsValidator.java

```diff
@@ -54,8 +54,10 @@
  * it is not part of the input schema and it is not outputted by any other operation occurring prior to
  * OP3, even though [z] is created by OP4 which occurs after OP3.
  *
- * For each operation, the output generated by it must be the part of the output schema or it must be
- * used as an input by the subsequent operations recorded by that stage.
+ * For each operation, the output generated by it is valid in the following cases:
+ * 1. The output is the part of the output schema,
+ * 2. The output is used as an input by the subsequent operations recorded by that stage.
+ * 3. The input fields of the operation of the output are all from the input schema of that stage.
```

Comment:
I don't understand that. So if A is in the input schema, then I rename A to B, then I create X from B and C, but X is not used afterwards, then it is invalid. But if X is created directly from A, then it is ok?